### PR TITLE
Remove unreachable code

### PR DESCRIPTION
**Pull request by practicalswift**
Remove unreachable code.

### CODE REVIEW

File: src/wallet/db.cpp [hunk 1]
```diff
@@ -554,7 +554,6 @@ bool CDB::Rewrite(CWalletDBWrapper& dbw, const char* pszSkip)
         }
         MilliSleep(100);
     }
-    return false;
```
Comment (promag) [inline src/wallet/db.cpp]: Mixed feelings here, maybe move up `bool fSuccess = true;` and break in the loop? Same below in `Backup()`?

Comment (TheBlueMatt) [inline src/wallet/db.cpp]: Agreed, though cant say I care *too* strongly. Its a smaller/simpler change not to.

**Comment by TheBlueMatt** [comment]
utACK 41bf1598f12d546851e509d369783d493bdd7108

File: src/wallet/db.cpp (hunk 1 above)
Comment (gmaxwell) [inline src/wallet/db.cpp]: promag's solution is a bit more robust against code being moved around, otoh, if the end becomes reachable and there is no return statement the compiler will warn...  though are we sure we won't get false positives for that warning on some compilers?

Comment (promag) [inline src/wallet/db.cpp]: Yes the compiler should warn. For me this is too long for not to have a clear return value. But consider it a nit.

Comment (promag) [inline src/wallet/db.cpp]: Maybe leave to a future PR to remove the `while (true)`.

**Pull request merged**